snapcraft/storeapi/v2/channel_map.py

```diff
@@ -0,0 +1,266 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List, Optional, Set, Union
+
+"""
+This module holds representations for results for the v2 channel-map
+API endpoint provided by the Snap Store.
+
+The full API is documented on
+https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map
+"""
+
+
+class Progressive:
+    """
+    Represents Progressive information for a MappedChannel.
+    """
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Union[str, bool]]) -> "Progressive":
```

Comment:
While entirely correct, I wonder if trying to maintain the `Union`s for (un)mashaling is a bit unnecessary instead of say `Any`.... but your call :-)

Reply:
if we need more, we can move to Any, but 2 values or 3 is still manageable